File: buy_stock/buy_stock.js
var maxProfit = function(prices) {
  let buy = 0;
  let sell = 1;
  let max_profit = 0;
  
  while (sell < prices.length){
    if (prices[buy] < prices[sell]) {
      let profit = prices[sell] - prices[buy];
      max_profit = Math.max(max_profit, profit);
    } else {
      buy = sell;
    }
    sell++;
  }
  return max_profit;
};

File: buy_stock/buy_stock.rb
def max_profit(prices)
  buy = 0
  sell = 1
  max_profit = 0

  while sell < prices.length do
    if prices[buy] < prices[sell]
      profit = prices[sell] - prices[buy]
      max_profit = [max_profit, profit].max
    else
      buy = sell
    end
    sell += 1
  end

  max_profit
end